File: mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    for (int i = 0; i < height; i++) //creates each row
    {
        //We want x spaces, where x is height - 1, height - 2, ...
        // we want y hashes, where y is 1, 2, up to height

        //spaces
        for (int j = height - 1; j > i; j--)
        {
            printf(" ");
        }

        //hashes right centered
        for (int h = 0; h <= i; h++)
        {
            printf("#");
        }

        printf("  ");

        //hashes left centered
        for (int h = 0; h <= i; h++)
        {
            printf("#");
        }

        printf("\n");
    }
}

File: mario.py
from cs50 import get_int
while True:
    height = get_int("Height: ")
    if height > 0 and height < 9:
        break

for i in range(height):
    # spaces
    for j in range(height - 1, i, -1):
        print(" ", end="")
    # Right aligned hashes
    for k in range(-1, i, 1):
        print("#", end="")

    print("  ", end="")

    # Left aligned hashes
    for h in range(-1, i, 1):
        print("#", end="")
    print()